src/overpass_api/statements/area_query.cc

```diff
@@ -287,8 +287,20 @@ void Area_Query_Statement::get_ranges
 
 bool Area_Constraint::delivers_data(Resource_Manager& rman)
 {
+  int counter = 0;
+
   if (!area->areas_from_input())
-    return false;
+  {
+    Block_Backend< Uint31_Index, Area_Skeleton > area_locations_db
+        (rman.get_area_transaction()->data_index(area_settings().AREAS));
+    for (Block_Backend< Uint31_Index, Area_Skeleton >::Flat_Iterator
+        it(area_locations_db.flat_begin());
+        !(it == area_locations_db.flat_end()); ++it)
+    {
+      if (area->get_submitted_id() == it.object().id.val())
```

Comment:
Then I get the following compile-time error (gcc 6.2):

> error: no match for ‘operator==’ (operand types are ‘long long int’ and ‘Area_Skeleton::Id_Type {aka Uint32_Index}’

Uint32_Index has the == operator implemented only for comparing with other Uint32_Index values while Area_Query_Statement (for some reason) stores the id it gets in a 64bit integer.

But I think that calling `.val()` here shouldn't be an issue, since the compiler can inline/optimize that away very easily.